src/clr-core/button/icon-button.element.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  baseStyles,
+  iconSlot,
+  iconSpinner,
+  iconSpinnerCheck,
+  property,
+  registerElementSafely,
+} from '@clr/core/internal';
+import { html } from 'lit-element';
+import { styles as baseButtonStyles } from './base-button.element.css.js';
+import { styles } from './icon-button.element.css.js';
+import { CdsButton, ClrLoadingState } from './button.element.js';
+
+/**
+ * Icon buttons give applications a compact alternative to communicate action and direct user intent.
+ *
+ * ```typescript
+ * import '@clr/core/icon-button';
+ * ```
+ *
+ * ```html
+ * <cds-icon-button><cds-icon shape="download"></cds-icon></cds-icon-button>
+ * ```
+ * @beta
+ * @element cds-icon-button
+ * @slot default - Content slot for inside the button
+ * @cssprop --background
+ * @cssprop --border-color
+ * @cssprop --border-radius
+ * @cssprop --border-width
+ * @cssprop --box-shadow-color
+ * @cssprop --color
+ * @cssprop --font-size
+ * @cssprop --height
+ * @cssprop --padding
+ */
+export class CdsIconButton extends CdsButton {
+  /**
+   * The aria-label attribute is required for accessibility. The cds-icon-button
+   * will warn if used without the aria-label being set.
+   *
+   * Ideally, the aria-label will be specific to the button's purpose. Avoid sharing
+   * generic labels across multiple icon buttons on a page.
+   */
+  @property({ type: String })
+  ariaLabel: string;
```

Comment:
For follow up pr, the warn/error feature of the property decorator is in master so we can start using it

Reply:
that's good to know. i'll need to use it in the alerts PR. so i'll make a note to add it here as well.

Reply:
actually, i have to do a squash before i merge. so i may as well add it now...